src/inference/emcee.jl

```diff
@@ -38,23 +38,24 @@ function AbstractMCMC.step(
 
     # Update the parameters if provided.
     if haskey(kwargs, :init_params)
-        for vi in vis
-            initialize_parameters!(vi, kwargs[:init_params], spl)
+        vis = map(vis) do vi
+            vi = initialize_parameters!!(vi, kwargs[:init_params], spl)
 
             # Update log joint probability.
-            model(rng, vi, SampleFromPrior())
+            last(DynamicPPL.evaluate!!(model, rng, vi, SampleFromPrior()))
         end
     end
 
     # Compute initial transition and states.
     transition = map(vis) do vi
         Transition(vi)
     end
+    # TODO: Make compatible with immutable `AbstractVarInfo`.
+    # Transform to unconstrained space.
+    DynamicPPL.link!.(vis, Ref(spl))
```

Comment:
Why did you extract it? It seems there are no performance gains but rather now there are both a broadcasting and a map operation whereas before we just used a single `map`?

Reply:
I extracted it because at first I intended to make it compatible with immutable `AbstractVarInfo`, but then I realized this shouldn't be done in this PR :upside_down_face: I'll revert this change I guess :+1: